### PR TITLE
Added AMQP Module option to use only one channel

### PR DESCRIPTION
**Pull request by Tekill**
This PR add new option which fix this issue: 
#4280

### CODE REVIEW

**Comment by Naktibalda** [comment]
Are there any disadvantages of using only one channel compared to using a new channel for each action?

**Comment by Tekill** [comment]
Current code does not delete old created channels so we have to keep in memory what:
1. Channel count is limited and probably reachable (standard 65535);
1. Every time spends CPU time and available RAM for rabbitmq decreases. Low available memory will cause memory alarm.

In previous commit I made mistake because rabbitmq may sent signal "channel_close" (e.g when to purge non-existent queue). Fortunately php-amqplib has build in channel manager and we can delegate that headache to him. Now it stable and may handle wrong module usage.
I ran tests with this fix (600 tests cases, 20+ queues) and it reduced total test execution time (from 12m to 10m) with rabbitmq memory usage (from 350mb to 1mb). I see only advantages in this optimisation but made this feature disabled by default. Old code will work as worked even after updating.

**Comment by Tekill** [comment]
Ideally, I would like this option to be enabled by default, but it takes more time to check any possible cases. Using a new channel for each action eliminate any side effects.

**Comment by Naktibalda** [comment]
@Tekill Please rebase on 2.3 branch to get the tests passing.

**Comment by Tekill** [comment]
@Naktibalda mission complete.

File: src/Codeception/Module/AMQP.php [hunk 1]
```diff
@@ -47,12 +49,13 @@
...
-        'port'     => '5672',
-        'vhost'    => '/',
-        'cleanup'  => true,
+        'host'             => 'localhost',
+        'username'         => 'guest',
+        'password'         => 'guest',
+        'port'             => '5672',
+        'vhost'            => '/',
+        'cleanup'          => true,
+        'only_one_channel' => false
```
Comment (DavertMik) [inline src/Codeception/Module/AMQP.php]: `single_channel` seems nicer to me.
Could you change it, please?

Comment (Tekill) [inline src/Codeception/Module/AMQP.php]: Yep, il do it at evening.

**Comment by Tekill** [comment]
@DavertMik option has been renamed.

**Comment by DavertMik** [comment]
Thanks!

**Pull request merged**